File: code_examples/3-Structured_Problem_Solving/10-Try_Catch/CPP/main.cpp
// Copyright (c) 2020 St. Mother Teresa HS All rights reserved.
//
// Created by: Mr. Coxall
// Created on: Sep 2020
// This program checks for a runtime error

#include <iostream>
#include <string>

int main() {
    // this function checks for a runtime error
    std::string integerAsString;

    // Input
    std::cout << "Enter an integer: ";
    getline(std::cin, integerAsString);

    // Process & Output
    try {
        int integerAsNumber = stoi(integerAsString);
        std::cout << "You entered: " << integerAsNumber << "." << std::endl;
    }
    catch (const std::invalid_argument &err)
    {
        std::cout << "The error was: " << err.what() << "." << std::endl;
    }
    std::cout << "Thanks for playing." << std::endl;

    std::cout << "\nDone." << std::endl;
}

File: code_examples/3-Structured_Problem_Solving/10-Try_Catch/Go/main.go
/**
 * Created by: Mr. Coxall
 * Created on: Sep 2020
 * This program checks for a runtime error
 */

package main

import (
	"fmt"
	"strconv"
)

func main() {
	// this function checks for a runtime error

	var integerAsString string

	// input
	fmt.Print("Enter an integer: ")
	fmt.Scanln(&integerAsString)

	// process and output
	integerAsNumber, err := strconv.Atoi(integerAsString)
	if err == nil {
		fmt.Println("You entered: ", integerAsNumber)
	} else {
		fmt.Println("The error was:", err, ".")
	}
	fmt.Println("Thanks for playing.")

	fmt.Println("\nDone.")
}
